lib/pkg.ml

```diff
@@ -262,32 +261,28 @@ let distrib_uri_of_homepage p =
   | Some (uri, _) ->
       path_of_distrib p >>| fun path -> Some (uri_append uri path)
 
-let distrib_uri ?(raw = false) p =
-  let subst_uri p uri =
-    name p >>= fun name ->
-    tag p >>= fun tag ->
-    let defs = String.Map.(empty |> add "NAME" name |> add "TAG" tag) in
-    Pat.of_string uri >>| fun pat -> Pat.format defs pat
-  in
-  let uri =
-    match p.distrib_uri with
-    | Some u -> Ok u
-    | None -> (
-        distrib_uri_of_homepage p >>= function
-        | Some u -> Ok u
-        | None -> (
-            distrib_uri_of_dev_repo p >>= function
-            | Some u -> Ok u
-            | None -> err_not_found () ) )
-  in
-  uri >>= fun uri ->
+let distrib_uri ?uri p =
```

Comment:
Just to confirm, this extra argument will go away once we move this function from `Pkg` to `Pkg.Github` right?

Reply:
Also it's probably better written directly at the call site as a `Option.value_lazy ~default:(fun () -> Pkg.distrib_uri pkg) distrib_uri`.

Reply:
For now this argument corresponds to the value of the `--distrib-uri` option, but I can rewrite the check fir this option earlier indeed.

Reply:
I removed the argument, let me know what you think